pkg/authn/oidc.go

```diff
@@ -34,12 +36,34 @@ type OIDCConfig struct {
 	SupportedSigningAlgs []string
 }
 
+type caContentProvider func() []byte
+
+func (f caContentProvider) CurrentCABundleContent() []byte {
+	return f()
+}
+
+func newCAContentProvider(filename string) (caContentProvider, error) {
+	pemBlock, err := ioutil.ReadFile(filename)
+	if err != nil {
+		return nil, err
+	}
+
+	return func() []byte {
+		return pemBlock
+	}, nil
+}
+
 // NewOIDCAuthenticator returns OIDC authenticator
 func NewOIDCAuthenticator(config *OIDCConfig) (authenticator.Request, error) {
+	cacp, err := newCAContentProvider(config.CAFile)
+	if err != nil {
+		return nil, err
+	}
+
 	tokenAuthenticator, err := oidc.New(oidc.Options{
 		IssuerURL:            config.IssuerURL,
 		ClientID:             config.ClientID,
-		CAFile:               config.CAFile,
+		CAContentProvider:    cacp,
```

Comment:
please use https://github.com/kubernetes/kubernetes/blob/release-1.23/staging/src/k8s.io/apiserver/pkg/server/dynamiccertificates/dynamic_cafile_content.go#L75 instead.

Reply:
using this will give us rotation of CA content for free.

Reply:
@s-urbaniak, PTAL at the latest commit.